admin_pages/messages/Messages_Admin_Page.core.php

```diff
@@ -4400,44 +4356,19 @@ public function save_settings()
         }
 
 
-        if ($this->_req_data['type'] === 'messenger') {
+        if ($type === 'messenger') {
             // this should be an array.
-            $settings  = $this->_req_data['messenger_settings'];
+            $settings  = $this->request->getRequestParam('messenger_settings', [], 'string', true);
             $messenger = $settings['messenger'];
-            // let's setup the settings data
-            foreach ($settings as $key => $value) {
-                switch ($key) {
-                    case 'messenger':
-                        unset($settings['messenger']);
-                        break;
-                    case 'message_types':
-                        unset($settings['message_types']);
-                        break;
-                    default:
-                        $settings[ $key ] = $value;
-                        break;
-                }
-            }
+            // remove messenger and message_types from settings array
+            unset($settings['messenger'], $settings['message_types']);
```

Comment:
👌

Reply:
wasn't that just the most bonkers silly code ever?
looping through an array, then using a switch to find a couple of specific keys, so that you can call `unset()` on those keys!
😱

Reply:
For sure !

Reply:
What a code! Thanks for fixing.